src/twisted/conch/ssh/keys.py

```diff
@@ -1198,8 +1343,19 @@ def sign(self, data):
             # Make sure they are padded out to 160 bits (20 bytes each)
             ret = common.NS(int_to_bytes(r, 20) + int_to_bytes(s, 20))
 
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+        elif keyType == 'EC': # pragma: no branch
```

Comment:
Could you remove the pragma statement? We don't add coverage exceptions to Twisted code.